dbms/src/Flash/Pipeline/Schedule/Events/LoadBucketEvent.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Pipeline/Schedule/Events/LoadBucketEvent.h>
+#include <Flash/Pipeline/Schedule/Tasks/LoadBucketTask.h>
+#include <Operators/SharedAggregateRestorer.h>
+
+namespace DB
+{
+std::vector<TaskPtr> LoadBucketEvent::scheduleImpl()
+{
+    assert(loader);
+    std::vector<TaskPtr> tasks;
+    auto load_inputs = loader->getNeedLoadInputs();
+    tasks.reserve(load_inputs.size());
+    for (const auto & input : load_inputs)
+        tasks.push_back(std::make_unique<LoadBucketTask>(mem_tracker, log->identifier(), exec_status, shared_from_this(), *input));
+    return tasks;
+}
+
+void LoadBucketEvent::finishImpl()
+{
+    assert(loader);
+    loader->storeBucketData();
```

Comment:
Why we need to store data in the `finishImpl`? When we should call `finishImpl`?

Reply:
LoadBucketEvent just reads the block from the stream, and then needs to move the block of the target bucket num from the input to `SharedBucketDataLoader::bucket_data_queue`

Reply:
> 

What will happen if we don't store data in `finishImpl`?